addons/gamification/models/karma.py

```diff
@@ -2,9 +2,28 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from werkzeug.exceptions import Forbidden
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class KarmaError(Forbidden):
     """ Karma-related error, used for forum and posts. """
     pass
+
+
+class KarmaRank(models.Model):
+    _name = 'gamification.karma.rank'
+    _description = 'Rank based on karma'
+    _order = 'minimum_value'
+
+    name = fields.Text(string='Rank Name')
+    description = fields.Text(string='Description')
+    minimum_value = fields.Integer(string='Minimal value', help='Minimum karma needed to reach this rank')
+    user_ids = fields.One2many('res.users', 'rank_id', string='Users', help="Users having this rank")
+    image = fields.Binary('Rank Icon')
+
+    @api.model_create_multi
+    def create(self, values_list):
+        res = super(KarmaRank, self).create(values_list)
+        users = self.env['res.users'].sudo().search([('karma', '>', 0)])
```

Comment:
Great, that way we don't update users with 0 karma (we have a lot of them at odoo). Good one.